samples/snippets/cloud-client/src/main/java/recaptcha/account_defender/AccountDefenderAssessment.java

```diff
@@ -0,0 +1,155 @@
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.UUID;
+
+public class AccountDefenderAssessment {
+
+  public static void main(String[] args) throws IOException, NoSuchAlgorithmException {
+    // TODO(developer): Replace these variables before running the sample.
+    // projectId: Google Cloud Project ID
+    String projectId = "project-id";
+
+    // recaptchaSiteKey: Site key obtained by registering a domain/app to use recaptcha
+    //    * services.
+    String recaptchaSiteKey = "recaptcha-site-key";
+
+    // token: The token obtained from the client on passing the recaptchaSiteKey.
+    String token = "recaptcha-token";
+
+    // recaptchaAction: Action name corresponding to the token.
+    String recaptchaAction = "recaptcha-action";
+
+    // Unique id of the customer (id can be email, customer id, etc.,).
+    String uniqueCustomerId = "default" + UUID.randomUUID().toString().split("-")[0];
+
+    // Hash the unique customer id using HMAC SHA-256.
+    MessageDigest digest = MessageDigest.getInstance("SHA-256");
+    byte[] hashBytes = digest.digest(uniqueCustomerId.getBytes(StandardCharsets.UTF_8));
+    ByteString hashedAccountId = ByteString.copyFrom(hashBytes);
+
+    accountDefenderAssessment(projectId, recaptchaSiteKey, token, recaptchaAction, hashedAccountId);
+  }
+
+  /**
+   * This assessment detects account takeovers. Input -> Pass in the hashed account id. Result ->
+   * Tells if the action represents an account takeover. You can optionally take actions (to trigger
+   * an MFA) based on the result.
+   */
+  public static void accountDefenderAssessment(
+      String projectId,
+      String recaptchaSiteKey,
+      String token,
+      String recaptchaAction,
+      ByteString hashedAccountId)
+      throws IOException {
+    try (RecaptchaEnterpriseServiceClient client = RecaptchaEnterpriseServiceClient.create()) {
+
+      // Set the properties of the event to be tracked.
+      Event event =
+          Event.newBuilder()
+              .setSiteKey(recaptchaSiteKey)
+              .setToken(token)
+              // Set the hashed account id (of the user).
+              // Recommended approach: HMAC SHA256 along with salt (or secret key).
+              .setHashedAccountId(hashedAccountId)
+              .build();
+
+      // Build the assessment request.
+      CreateAssessmentRequest createAssessmentRequest =
+          CreateAssessmentRequest.newBuilder()
+              .setParent(ProjectName.of(projectId).toString())
+              .setAssessment(Assessment.newBuilder().setEvent(event).build())
+              .build();
+
+      Assessment response = client.createAssessment(createAssessmentRequest);
+
+      // Check integrity of the response token.
+      if (!checkTokenIntegrity(response.getTokenProperties(), recaptchaAction)) {
+        return;
```

Comment:
This return on failure is unnoticeable by the caller.

Reply:
checkTokenIntegrity prints the message upon returning false.

Reply:
I am not sure that I'm following this comment. The checkTokenIntegrity() method prints a specific message upon failure. 
Do you also want to print a message for success?

Reply:
Not a big problem. I would put the print statement here at line 100, just before return.